lms/djangoapps/grades/signals/handlers.py

```diff
@@ -124,21 +113,32 @@ def recalculate_subsection_grade_handler(sender, **kwargs):  # pylint: disable=u
        - usage_id: Unicode string indicating the courseware instance
     """
     try:
-        course_id = kwargs.get('course_id', None)
-        usage_id = kwargs.get('usage_id', None)
-        student = kwargs.get('user', None)
+        course_id = kwargs['course_id']
+        usage_id = kwargs['usage_id']
+        student = kwargs['user']
+    except KeyError:
+        log.exception(
```

Comment:
I really like this refactor. I guess if the usage or course id's passed in can't be converted to keys, we'll just bubble up that uncaught exception, right?